admin-dev/themes/new-theme/js/pages/order/create/cart-rule-searcher.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Router from '../../../components/router';
+import createOrderMap from './create-order-map';
+import CartRulesRenderer from './cart-rules-renderer';
+
+const $ = window.$;
+
+/**
+ * Responsible for searching cart rules and managing cart rules search block
+ */
+export default class CartRuleSearcher {
+  constructor() {
+    this.router = new Router();
+    this.$searchInput = $(createOrderMap.cartRuleSearchInput);
+    this.$searchResultBox = $(createOrderMap.cartRulesSearchResultBox);
+    this.cartRulesRenderer = new CartRulesRenderer();
+
+    return {
+      onCartRuleSearch: () => {
+        this._search();
+      },
+      onCartRuleSelect: (cartRuleId, cartId) => {
+        this._addCartRuleToCart(cartRuleId, cartId);
+      },
+      onDoneSearchingCartRule: () => {
+        this._hideResultsDropdown();
+      },
+    };
+  }
+
+  /**
+   * Searches for cart rules by search phrase
+   *
+   * @private
+   */
+  _search() {
+    const searchPhrase = this.$searchInput.val();
+    if (searchPhrase.length < 3) {
```

Comment:
@matks, should i also restrict this in php side handler?

Reply:
Not necessary, depends if you're sure it's a global behaviour or only for the front.